src/Entities/Entity.cpp

```diff
@@ -1080,57 +1080,126 @@ void cEntity::HandlePhysics(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 
 	NextSpeed += m_WaterSpeed;
 
-	if (NextSpeed.SqrLength() > 0.0f)
-	{
-		Vector3d HitCoords;
-		Vector3i HitBlockCoords;
-		eBlockFace HitBlockFace;
-		Vector3d wantNextPos = NextPos + NextSpeed * DtSec.count();
-		auto isHit = cLineBlockTracer::FirstSolidHitTrace(*GetWorld(), NextPos, wantNextPos, HitCoords, HitBlockCoords, HitBlockFace);
-		if (isHit)
+	if (NextSpeed.HasNonZeroLength())
+	{
+		Vector3d SizeLimit[]
 		{
-			// Set our position to where the block was hit, minus a bit:
-			// TODO: The real entity's m_Width should be taken into account here
-			NextPos = HitCoords - NextSpeed.NormalizeCopy() * 0.1;
-			if (HitBlockFace == BLOCK_FACE_YP)
-			{
-				// We hit the ground, adjust the position to the top of the block:
-				m_bOnGround = true;
-				NextPos.y = HitBlockCoords.y + 1;
-			}
+			{ m_Width, m_Height, m_Width },
+			{ m_Width, 0, 0 },
+			{ 0, m_Height, 0 },
+			{ 0, 0, m_Width },
+		};
 
-			// Avoid movement in the direction of the blockface that has been hit:
-			switch (HitBlockFace)
+		for (unsigned long i = 0; i < ARRAYCOUNT(SizeLimit); i++)
+		{
+			switch (i)
 			{
-				case BLOCK_FACE_XM:
-				case BLOCK_FACE_XP:
+				case 0:
```

Comment:
This just looks wrong. It is like putting a sequential program into a loop:

```cpp
for(int i = 0; i < 3; i++)
{
    switch(i)
    {
        case 0: FirstInstruction; break;
        case 1: SecondInstruction; break;
        case 2: ThirdInstruction; break;
    }
}
```
instead of
```cpp
FirstInstruction;
SecondInstruction;
ThirdInstruction;
```
And if somebody changes the loop counter, we may have a hard day...